src/settings.js

```diff
@@ -227,23 +228,31 @@ parseCrop = function(val) {
 }
 
 parseShrink = function(val) {
-  var reShrink, shrink, size
-  reShrink = /^([0-9]+)x([0-9]+)(?:\s+(\d{1,2}|100)%)?$/i
-  shrink = reShrink.exec($.trim(val.toLowerCase())) || []
+  const reShrink = /^([0-9]+)x([0-9]+)(?:\s+(\d{1,2}|100)%)?$/i
+  const shrink = reShrink.exec($.trim(val.toLowerCase())) || []
   if (!shrink.length) {
     return false
   }
-  size = shrink[1] * shrink[2]
-  if (size > 5000000) {
-    // ios max canvas square
+  const size = shrink[1] * shrink[2]
+  if (size > MAX_SQUARE_SIDE * MAX_SQUARE_SIDE) {
+    // Synthetic limit of 75 Mpx
     warnOnce(
-      'Shrinked size can not be larger than 5MP. ' +
+      'Shrinked size can not be larger than 75MP. ' +
         `You have set ${shrink[1]}x${shrink[2]} (` +
         `${Math.ceil(size / 1000 / 100) / 10}MP).`
     )
 
     return false
   }
+  if (shrink[1] > MAX_SIDE || shrink[2] > MAX_SIDE) {
+    // Synthetic limit of 16384 pixels
```

Comment:
I have defined a synthetic limits on the image shrink size. It's 75 Mpx for square area and 16384 px for square dimension.

The main reason is performance. Modern browsers have up to 16384*16384 square area limit (webkit), and up to 8388607px dimension limit (safari). Allocating such huge canvases is too slow (it takes > 150ms on my laptop). And it's unnecessary to test, because this is too much for image shrink.

So, 75 Mpx square and 16384px side should be more than enough for image shrink.